common-pg/src/main/scala/com/socrata/pg/store/PGSecondaryDatasetMapReader.scala

```diff
@@ -2,52 +2,155 @@ package com.socrata.pg.store
+      using(t("rollup", "copy_id" -> copyInfo.systemId, "name" -> name)(stmt.executeQuery())) { rs =>
+        if (rs.next()) {
+          Some(new LocalRollupInfo(copyInfo, name, rs.getString("soql"), rs.getString("table_name"), rs.getLong("system_id")))
         } else {
           None
         }
       }
     }
   }
+
+  def getRollupCopiesRelatedToCopyQuery =
+    """select dataset_map.system_id           as dataset_map_system_id,
+       dataset_map.next_counter_value  as dataset_map_next_counter_value,
+       dataset_map.latest_data_version as dataset_map_latest_data_version,
+       dataset_map.locale_name         as dataset_map_locale_name,
+       dataset_map.obfuscation_key     as dataset_map_obfuscation_key,
+       dataset_map.resource_name       as dataset_map_resource_name,
+       copy_map.system_id as copy_map_system_id,
+       copy_map.copy_number as copy_map_copy_number,
+       copy_map.lifecycle_stage :: TEXT as copy_map_lifecycle_stage,
+       copy_map.data_version as copy_map_data_version,
+       copy_map.data_shape_version as copy_map_data_shape_version,
+       copy_map.last_modified as copy_map_last_modified,
+       copy_map_table_modifiers.table_modifier as copy_map_table_modifiers_table_modifier
+from rollup_relationship_map
+         join rollup_map on rollup_relationship_map.rollup_system_id = rollup_map.system_id
+         join copy_map on rollup_map.copy_system_id = copy_map.system_id
+         left join copy_map_table_modifiers ON copy_map.system_id = copy_map_table_modifiers.copy_system_id
+         join dataset_map on copy_map.dataset_system_id = dataset_map.system_id
+where referenced_copy_system_id = ?"""
+
+  def getRollupCopiesRelatedToCopy(copyInfo: CopyInfo): Set[CopyInfo] = {
+    using(conn.prepareStatement(getRollupCopiesRelatedToCopyQuery)) { stmt =>
+      stmt.setLong(1, copyInfo.systemId.underlying)
+      using(t("getRollupCopiesRelatedToCopy", "copy_id" -> copyInfo.systemId)(stmt.executeQuery())) { rs =>
+        Iterator
+          .continually(rs.next)
+          .takeWhile(identity)
+          .map(_ => {
+            CopyInfo(
+              DatasetInfo(new DatasetId(rs.getLong("dataset_map_system_id")), rs.getLong("dataset_map_next_counter_value"), rs.getLong("dataset_map_latest_data_version"), rs.getString("dataset_map_locale_name"), rs.getBytes("dataset_map_obfuscation_key"), Option(rs.getString("dataset_map_resource_name"))),
+              new CopyId(rs.getLong("copy_map_system_id")),
+              rs.getLong("copy_map_copy_number"),
+              LifecycleStage.valueOf(rs.getString("copy_map_lifecycle_stage")),
+              rs.getLong("copy_map_data_version"),
+              rs.getLong("copy_map_data_shape_version"),
+              new DateTime(rs.getTimestamp("copy_map_last_modified").getTime),
+              Some(rs.getLong("copy_map_table_modifiers_table_modifier"))
+            )
+          })
+          .toSet
+      }
```

Comment:
I'm not a huge fan of this kind of interleaved side-effects on iterators.  I prefer to just embrace the imperativity and use a while loop.